peerdid/peer_did.py

```diff
@@ -24,23 +28,60 @@ def create_peer_did_numalgo_0(inception_key: PublicKeyAuthentication) -> PEER_DI
     (https://identity.foundation/peer-did-method-spec/index.html#generation-method)
     For this type of algorithm did_doc can be obtained from peer_did
     :param inception_key: the key that creates the DID and authenticates when exchanging it with the first peer
+    :raises TypeError: if the inception_key is not instance of PublicKeyAuthentication
+    :raises ValueError: if the inception_key is not instance base58 encoded
     :return: generated peer_did
     """
-    pass
+    if not isinstance(inception_key, PublicKeyAuthentication):
+        raise TypeError(f"Wrong type of inception_key: {str(type(inception_key))}. Expected: PublicKeyAuthentication")
+    if not _check_if_base58_encoded(inception_key.encoded_value):
```

Comment:
The key is not necessary in base58 encoding. It's just one of possible options (though the only one we support right now). So, it's better to rename it to `_check_key_encoding` which will check the encoding and raise a corresponding error text based on the encoding type enum.